Algorithms/Numeric/Factorial.cs

```diff
@@ -9,18 +10,36 @@ namespace Algorithms.Numeric
     public static class Factorial
     {
         /// <summary>
-        ///     Calculates factorial of a number.
+        ///     Calculates factorial of a integer number.
         /// </summary>
-        /// <param name="num">Input number.</param>
-        /// <returns>Factorial of input number.</returns>
-        public static long Calculate(int num)
+        /// <param name="inputNum">Integer Input number.</param>
+        /// <returns>Factorial of integer input number.</returns>
+        public static BigInteger Calculate(int inputNum)
         {
-            if (num < 0)
+            // Convert integer input to BigInteger
+            BigInteger num = new BigInteger(inputNum);
+
+            // Don't calculate factorial if input is a negative number.
+            if (BigInteger.Compare(num, BigInteger.Zero) < 0)
             {
                 throw new ArgumentException("Only for num >= 0");
             }
 
-            return num == 0 ? 1 : num * Calculate(num - 1);
+            // Factorial of 0 is 1. No need to calculate further.
+            if(BigInteger.Compare(num, BigInteger.Zero) == 0)
+            {
+                return BigInteger.One;
+            }
```

Comment:
This check is redundant, the loop below gives the same result

Reply:
Requested change is implemented.